nutrition_labels/grant_tagger.py

```diff
@@ -80,8 +57,36 @@ def transform(self, data):
             scaler = MinMaxScaler()
             X_vect = scaler.fit_transform(X_vect)
 
+        return X_vect, y
+
+    def split_data(self, X_vect,y, sample_not_relevant, irrelevant_sample_seed,split_seed):
+        self.sample_not_relevant = sample_not_relevant
+
+        relevant_sample_index = [ind for ind,x in enumerate(y) if x != 0]
+        irrelevant_sample_index = [ind for ind, x in enumerate(y) if x == 0]
+        irrelevant_sample_size = int(round((len(relevant_sample_index) * self.sample_not_relevant)))
+
+        if not self.sample_not_relevant and self.sample_not_relevant != 0 :
+            # If you don't specify sample_not_relevant
+            # then use all the not relevant data points
+            sample_size = len(irrelevant_sample_index)
+        else:
+            # If the inputted sample size is larger than the number of
+            # data points, then just use all the data points
+            sample_size = min(len(irrelevant_sample_index), irrelevant_sample_size)
+
+        if sample_size < len(irrelevant_sample_index):
+            seed(irrelevant_sample_seed)
+            sample_index = relevant_sample_index + sample(irrelevant_sample_index,sample_size)
+            y = y[sample_index]
+            if self.vectorizer_type == 'bert':
+                X_vect = [X_vect[i] for i in sample_index]
+            else:
+                X_vect = X_vect[sample_index]
```

Comment:
I think now you can get rid of the `sample_size ` parameters and replace all this with:
```
if sample_not_relevant < 1:
   relevant_sample_index = [ind for ind,x in enumerate(y) if x != 0]
   irrelevant_sample_index = [ind for ind, x in enumerate(y) if x == 0]
   irrelevant_sample_size = int(round((len(irrelevant_sample_index) * self.sample_not_relevant)))
   seed(irrelevant_sample_seed)
   sample_index = relevant_sample_index + sample(irrelevant_sample_index, irrelevant_sample_size)
   y = y[sample_index]
   if self.vectorizer_type == 'bert':
          X_vect = [X_vect[i] for i in sample_index]
    else:
          X_vect = X_vect[sample_index]
```

Reply:
after our conversation about `sample_not_relevant` being a ratio the above comment is mostly wrong, I think it can now be:
```
relevant_sample_index = [ind for ind,x in enumerate(y) if x != 0]
irrelevant_sample_index = [ind for ind, x in enumerate(y) if x == 0]
irrelevant_sample_size = int(round((len(relevant_sample_index) * self.sample_not_relevant)))

if irrelevant_sample_size < len(irrelevant_sample_index):
            seed(irrelevant_sample_seed)
            sample_index = relevant_sample_index + sample(irrelevant_sample_index, irrelevant_sample_size)
            y = y[sample_index]
            if self.vectorizer_type == 'bert':
                X_vect = [X_vect[i] for i in sample_index]
            else:
                X_vect = X_vect[sample_index]
```